$size combined with $all gives incorrect results

Here's a MWE where I've found some disagreement between mongomock (3.19) and pymongo (3.10.1) when using the `$all` and `$size` together on an array.

```python
import mongomock
import pymongo

docs = [
    {"values": ["Abcd"]},
    {"values": ["Abcd", "Bcdef"]},
    {"values": ["Abcd", "Bcdef", "Cdefg"]},
]

mm_db = mongomock.MongoClient().db.coll
mm_db.drop()
mm_db.insert_many(docs)
pm_db = pymongo.MongoClient().db.coll
pm_db.drop()
pm_db.insert_many(docs)

# works fine
query = {"values": {"$all": ["Abcd"]}}
pm_curs = list(pm_db.find(query))
mm_curs = list(mm_db.find(query))
assert (
    len(pm_curs) == len(mm_curs) == 3
), f"mm returned {len(mm_curs)} results vs pm {len(pm_curs)}"

# works fine
query = {"values": {"$size": 1}}
pm_curs = list(pm_db.find(query))
mm_curs = list(mm_db.find(query))
assert (
    len(pm_curs) == len(mm_curs) == 1
), f"mm returned {len(mm_curs)} results vs pm {len(pm_curs)}"

# fails, mm returns all 3 documents
query = {"values": {"$size": 1, "$all": ["Abcd"]}}
pm_curs = list(pm_db.find(query))
mm_curs = list(mm_db.find(query))
assert (
    len(pm_curs) == len(mm_curs) == 1
), f"mm returned {len(mm_curs)} results vs pm {len(pm_curs)}"

# works
query = {"values": {"$all": ["Abcd", "Bcdef"]}}
pm_curs = list(pm_db.find(query))
mm_curs = list(mm_db.find(query))
assert (
    len(pm_curs) == len(mm_curs) == 2
), f"mm returned {len(mm_curs)} results vs pm {len(pm_curs)}"


# fails, mm returns all 3 again
query = {"values": {"$size": 2, "$all": ["Abcd", "Bcdef"]}}
pm_curs = list(pm_db.find(query))
mm_curs = list(mm_db.find(query))
assert (
    len(pm_curs) == len(mm_curs) == 1
), f"mm returned {len(mm_curs)} results vs pm {len(pm_curs)}"
```

Output:
```
$ python --version
Python 3.7.6

$ python mongomock_test.py
Traceback (most recent call last):
  File "mongomock_test.py", line 29, in <module>
    ), f"mm returned {len(mm_curs)} results vs pm {len(pm_curs)}"
AssertionError: mm returned 3 results vs pm 1
```

Happy to try to fix this myself if it's a genuine bug!

--- a/mongomock/filtering.py
+++ b/mongomock/filtering.py
@@ -100,7 +100,9 @@ class _Filterer(object):
             if isinstance(search, dict) and '$all' in search:
                 if not self._all_op(iter_key_candidates(key, document), search['$all']):
                     return False
-                continue
+                # if there are no query operators then continue
+                if len(search) == 1:
+                    continue
 
             for doc_val in iter_key_candidates(key, document):
                 has_candidates |= doc_val is not NOTHING

--- a/tests/test__mongomock.py
+++ b/tests/test__mongomock.py
@@ -471,6 +471,15 @@ class MongoClientCollectionTest(_CollectionComparisonTest):
         self.cmp.compare.find({'_id': id, 'l_string': {'$not': {'$size': 0}}})
         self.cmp.compare.find({'_id': id, 'l_tuple': {'$size': 2}})
 
+    def test__size_with_all(self):
+        objs = [{'list': ['a']}, {'list': ['a', 123]}, {'list': ['a', 123, 'xyz']}]
+        self.cmp.do.insert_many(objs)
+        self.cmp.compare.find({'list': {'$all': ['a']}})
+        self.cmp.compare.find({'list': {'$all': ['a', 123]}})
+        self.cmp.compare.find({'list': {'$all': ['a'], '$size': 1}})
+        self.cmp.compare.find({'list': {'$all': ['a', 123], '$size': 2}})
+        self.cmp.compare.find({'list': {'$all': ['a', 123, 'xyz'], '$size': 3}})
+
     def test__regex_match_non_string(self):
         id = ObjectId()
         self.cmp.do.insert({

--- a/tests/test__mongomock.py
+++ b/tests/test__mongomock.py
@@ -471,14 +471,16 @@ class MongoClientCollectionTest(_CollectionComparisonTest):
         self.cmp.compare.find({'_id': id, 'l_string': {'$not': {'$size': 0}}})
         self.cmp.compare.find({'_id': id, 'l_tuple': {'$size': 2}})
 
-    def test__size_with_all(self):
+    def test__all_with_other_operators(self):
         objs = [{'list': ['a']}, {'list': ['a', 123]}, {'list': ['a', 123, 'xyz']}]
         self.cmp.do.insert_many(objs)
-        self.cmp.compare.find({'list': {'$all': ['a']}})
-        self.cmp.compare.find({'list': {'$all': ['a', 123]}})
         self.cmp.compare.find({'list': {'$all': ['a'], '$size': 1}})
         self.cmp.compare.find({'list': {'$all': ['a', 123], '$size': 2}})
         self.cmp.compare.find({'list': {'$all': ['a', 123, 'xyz'], '$size': 3}})
+        self.cmp.compare.find({'list': {'$all': ['a'], '$size': 3}})
+        self.cmp.compare.find({'list': {'$all': ['a', 123], '$in': ['xyz']}})
+        self.cmp.compare.find({'list': {'$all': ['a', 123, 'xyz'], '$in': ['abcdef']}})
+        self.cmp.compare.find({'list': {'$all': ['a'], '$eq': ['a']}})
 
     def test__regex_match_non_string(self):
         id = ObjectId()

